### PR TITLE
Make abyss mandatory

### PR DESCRIPTION
**Pull request by michael-loki**
Could maybe do with some more polishing, I'm not sure.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ install (root):
 
 build (where `$ARCH` is your platform - `i686` or `x86_64`):
 
-    $ pacman -Sy base-devel mingw-w64-$ARCH-toolchain git libtool autoconf mingw-w64-$ARCH-cmake
+    $ pacman -Sy base-devel mingw-w64-$ARCH-toolchain git libtool autoconf mingw-w64-$ARCH-cmake mingw-w64-$ARCH-rapidjson
```
Comment (despair86) [inline readme.md]:
```
[Ricardo@NT-CROSS-DEV projectrebirth]$ pacman -S mingw-w64-i686-rapidjson
resolving dependencies...
looking for conflicting packages...

Packages (1) mingw-w64-i686-rapidjson-1.1.0-1

Total Download Size:    1.26 MiB
Total Installed Size:  15.00 MiB

:: Proceed with installation? [Y/n] n
```
verified

**Comment by michael-loki** [comment]
re-spot-checked

File: CMakeLists.txt [hunk 2]
```diff
@@ -30,6 +30,7 @@ endif()
 if(WIN32)
   set(CMAKE_CXX_STANDARD 17)
   ENABLE_LANGUAGE(RC)
+  add_compile_options(-Wno-format)
```
Comment (despair86) [inline CMakeLists.txt]: don't need this anymore

**Comment by despair86** [comment]
looks good otherwise, building now

File: llarp/util/bencode.cpp [hunk 3]
```diff
@@ -75,7 +76,7 @@ bencode_write_bytestring(llarp_buffer_t* buff, const void* data, size_t sz)
 bool
 bencode_write_uint64(llarp_buffer_t* buff, uint64_t i)
 {
-  if(!buff->writef("i%lu", i))
+  if(!buff->writef("i%" PRIu64, i))
```
Comment (despair86) [inline llarp/util/bencode.cpp]: wasn't this supposed to fix your macintosh compile error?

**Pull request merged**